### ISSUE TITLE
Conscrypt library is better as separate apk

### ISSUE THREAD

**Issue by ale5000-git** [microg/android_packages_apps_GmsCore#802]
Conscrypt library alone (also with maximum compression) is bigger then the entire GmsCore, I think it would be better as separate apk.
As bonus it could be put on F-Droid and updated without updateing GmsCore.

@mar-v-in
@voidstarstar

**Comment by voidstarstar** [comment]
I do like the idea of making microG more modular so we can get faster security updates, but I'm not really sure how it would be implemented.

I didn't test it, but I wonder if upgrading to 2.1.0 will fix #759.  The [server it connects to](https://github.com/microg/android_packages_apps_GmsCore/issues/759#issuecomment-487155329) uses TLS 1.3.

Regarding the armeabi architecture, that sounds like it should be opened as a feature request for conscrypt.  I wasn't aware that they didn't support armeabi for their native libraries, but it might be an easy fix (e.g. a simple addition to their `build.gradle` line that lists the Android NDK architectures).

**Comment by mar-v-in** [comment]
Everything you said sounds like good ideas to me. When conscrypt is packaged through F-Droid default repository it wouldn't be picked up by the original play services "ProviderInstaller" automatically, but I can also imagine to provide a associated "client library" that would be able to do so, as this could be advantageous to many apps, even those that want to be free software.

Designing microG more modular while at the same time make it easier to develop, was on my list of todos anyway.

**Comment by Nanolx** [comment]
@voidstarstar nope, bumping conscrypt to 2.1.0 does not fix the issue with Discord, I've already tested that.

**Comment by voidstarstar** [comment]
#564 might also benefit from a more modular design.

@Nanolx Oh, that's a shame, but thanks for testing it!
